Systems/c172p-engine.xml

```diff
@@ -62,7 +62,121 @@
+                <ifthen>
+                    <eq>
+                        <p>/engines/active-engine/damage_allowed</p>
+                        <v>1</v>
+                    </eq>
+                    <sum>
+                        <property>engine/damage-level</property>
+                        <product>
+                           <table>
+                                <independentVar lookup="row">engine/rpm-over-redline</independentVar>
+                                <independentVar lookup="column">engine/oil-temp-above-max</independentVar>
+                                <tableData>
+                                         0     50    100    250
+                                    0    0      1      9     16
+                                  100    1      2     16     50
+                                  200    4      5     25     65
+                                  300    9     10     36     82
+                                  400   16     17     49    101
+                                  500   25     25     64    122
+                                  600   36     37     81    145
+                                  700   49     50    100    170
+                                  800   64     65    121    197
+                                  900   81     82    144    226
+                                 1000  100    101    169    256
+                                 1100  121    123    196    289
+                                 1200  144    147    225    324
+                                 1300  169    175    256    361
+                                 1400  196    199    289    400
+                                 1500  225    228    324    441
+                                </tableData>
+                            </table>
+                            <property>simulation/channel-dt</property>
+                        </product>
+                    </sum>
+                    <v>0</v> <!-- this is the value if damage is disabled -->
+                </ifthen>
+            </function>
+        </fcs_function>
+
+        <fcs_function>
+            <function> <!-- derived from a graphical calculator: 0.01x pow 2 + 2.15x + 0.625 multiplied by 2.4 -->
+                <product>
+                    <sum>
+                        <product>
+                            <v>0.00008</v>
+                            <pow>
+                                <property>engine/damage-level</property>
+                                <v>2</v>
+                            </pow>
+                        </product>
+                        <v>0.625</v>
+                    </sum>
+                    <v>2.4</v>
+                </product>
+            </function>
+            <clipto>
+                <min>0.001</min>
+                <max>10000</max>
+            </clipto>
+            <output>engine/friction-hp</output>
+        </fcs_function>
+
+        <switch name="propulsion/engine[0]/friction-hp">
```

Comment:
This did not work as intended unfortunately. So you would have to change it back to `name=engine/engine-0-friction-hp` and add `<output>propulsion/engine[0]/friction-hp</output>`.

Same for the other engine.